File: 백준/Silver/10828. 스택/스택.py
import sys
n = int(sys.stdin.readline().strip())
stack = []

def push(num):
    stack.append(num)
    
def pop():
    if stack:
        num = stack.pop()
        print(num) 
        
    else: 
        print('-1')

def size():
    print(len(stack))
    
def empty():
    print('1') if not stack else print(0)
    
def top():
    print(stack[-1]) if stack else print(-1)
    
for i in range(n):
    operation = list(map(str, sys.stdin.readline().split()))
    
    if len(operation) == 2:
        push(operation[1])
        
    else:
        if operation[0] == 'pop':
            pop()
            
        elif operation[0] == 'size':
            size()
            
        elif operation[0] == 'empty':
            empty()
            
        else:
            top()

File: 백준/Silver/10828. 스택/스택.java
import java.io.*;
import java.util.Stack;

public class Main {
    static StringBuilder sb = new StringBuilder();
    static Stack<Integer> stack = new Stack<>();

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());

        while (n-- > 0) {
            String operation = br.readLine();
            stackOperation(operation);
        }

        System.out.println(sb);
        br.close();
    }

    public static void stackOperation(String operation) {
        String[] o = operation.split(" ");
        if (o[0].equals("push")) {
            int num = Integer.parseInt(o[1]);
            stack.push(num);
        } else {
            int res;
            switch (o[0]) {
                case "pop":
                    res = stack.isEmpty()? -1 : stack.pop();
                    sb.append(res).append("\n");
                    break;

                case "size":
                    res = stack.size();
                    sb.append(res).append("\n");
                    break;

                case "empty":
                    res = stack.isEmpty()? 1 : 0;
                    sb.append(res).append("\n");
                    break;

                default:
                    res = stack.isEmpty()? -1 : stack.peek();
                    sb.append(res).append("\n");
                    break;
            }
        }
    }
}